hrjobcontract/CRM/Hrjobcontract/BAO/HRJobContractRevision.php

```diff
@@ -210,4 +236,84 @@ protected static function clearEffectiveEndDate($revisionId) {
     );
     CRM_Core_DAO::executeQuery($clearEffectiveEndDateQuery, $clearEffectiveEndDateParams);
   }
+
+  /**
+   * Builds the SQL query for fetching the revision details of the given entities
+   *
+   * The fields of each entities are dynamically listed in the SELECT clause
+   * and are aliased with a prefix ({entity}__) so that it is possible to know
+   * which entity they belong to (civicrm_hrjobcontract_pay.id -> pay__id)
+   *
+   * @param Array $entities
+   * @param Array $revision
+   */
+  protected static function buildFullDetailsQuery($entities, $revision) {
+    $query = ['select' => [], 'from' => [], 'where' => []];
+
+    /**
+     * Creates an alias for the given entity
+     *
+     * @param string $entity
+     * @return  string
+     */
+    $entityAlias = function ($entity) {
+      return substr($entity, 0, 3);
+    };
+
+    foreach ($entities as $entity) {
+      $class = "CRM_Hrjobcontract_BAO_HRJob" . ucfirst($entity);
+      $fields = array_column($class::fields(), 'name');
+
+      foreach ($fields as $field) {
+        $query['select'][] = "{$entityAlias($entity)}.{$field} as {$entity}__{$field}";
+      }
+
+      $query['from'][] = "civicrm_hrjobcontract_{$entity} {$entityAlias($entity)}";
+      $query['where'][] = "{$entityAlias($entity)}.jobcontract_revision_id = " . $revision["{$entity}_revision_id"];
+    }
+
+    return sprintf(
+      "SELECT %s FROM %s WHERE %s ",
+      join(', ', $query['select']),
+      join(', ', $query['from']),
+      join(' AND ', $query['where'])
+    );
+  }
+
+  /**
+   * Takes a flat list of revision details and rearranges it, by grouping
+   * each field by the entity it belongs to
+   *
+   * The field's entity is inferred by the prefix ({entity}__) in the field's name
+   *
+   * @param  Array $result
+   * @return Array
+   */
+  protected static function normalizeFullDetailsResult($result) {
+    $normalized = [];
+
+    foreach ($result as $key => $value) {
+      if (strpos($key, '_') == 0) { continue; } // ignores "internal" fields
```

Comment:
Instead of strpos, you can do something like this:

```php
if($key[0] == '_')
```

It's faster and more secure, since you won't have to mind about strpos which can return 0 and false